proxyd/consensus_poller.go

```diff
@@ -267,7 +275,12 @@ func (cp *ConsensusPoller) UpdateBackendGroupConsensus(ctx context.Context) {
 			continue
 		}
 
-		if lowestBlock == 0 || backendLatestBlockNumber < lowestBlock {
+		// find the highest common ancestor, ignoring backends that are too far lagging behind
+		// when the backend is too far ahead from current lowest, the current lowest is ignored
+		// when the backend if too far behind, the backend itself is ignored
+		if lowestBlock == 0 ||
```

Comment:
Just for my understanding here:

Let's say we have three nodes A, B, and C. A's height is 50, B's height is 100, and C's height is 101. Assume a max block lag of 10. If iterated in order A, B, and C:

1. For A, `lowestBlock` will first be set to 50.
2. For B, `lowestBlock` will be set to 100 because the`backendLatestBlockNumber > lowestBlock && uint64(backendLatestBlockNumber-lowestBlock) > cp.maxBlockLag` check will succeed.
3. For C, `lowestBlock` will be set to 100 because the `backendLatestBlockNumber > lowestBlock && uint64(backendLatestBlockNumber-lowestBlock) > cp.maxBlockLag` check will fail.

Reply:
Your understanding is correct